tests/unittests/plotting/test_plotly_backend.py

```diff
@@ -0,0 +1,97 @@
+import pytest
+import plotly
+
+from orion.plotting import regret
+from orion.core.utils.tests import create_experiment
+from orion.core.worker.experiment import ExperimentView
+
+config = dict(
+    name='experiment-name',
+    space={'x': 'uniform(0, 200)'},
+    metadata={'user': 'test-user',
+              'orion_version': 'XYZ',
+              'VCS': {"type": "git",
+                      "is_dirty": False,
+                      "HEAD_sha": "test",
+                      "active_branch": None,
+                      "diff_sha": "diff"}},
+    version=1,
+    pool_size=1,
+    max_trials=10,
+    working_dir='',
+    algorithms={'random': {'seed': 1}},
+    producer={'strategy': 'NoParallelStrategy'},
+)
+
+trial_config = {
+    'experiment': 0,
+    'status': 'completed',
+    'worker': None,
+    'start_time': None,
+    'end_time': None,
+    'heartbeat': None,
+    'results': [],
+    'params': []
+}
+
+
+def assert_regret_plot(plot):
+    assert plot.layout.title.text == "Regret for experiment 'experiment-name'"
+    assert plot.layout.xaxis.title.text == "Trials ordered by suggested time"
+    assert plot.layout.yaxis.title.text == "Objective 'loss'"
+
+    trace1 = plot.data[0]
+    assert trace1.type == "scatter"
+    assert trace1.name == "trials"
+    assert trace1.mode == "markers"
+    assert len(trace1.y) == 1
+    assert not trace1.x
+
+    trace2 = plot.data[1]
+    assert trace2.type == "scatter"
+    assert trace2.name == "best-to-date"
+    assert trace2.mode == "lines"
+    assert len(trace2.y) == 1
+    assert not trace2.x
+
+
```

Comment:
I think we could automate most of these tests across backends with pytest.mark.parametrize: https://docs.pytest.org/en/stable/parametrize.html

Reply:
That would be possible if there is a common format we could check the plots against or make test adapters for each backend.

Reply:
I think the adapter solution is the most elegant and allows us to define exactly what test information (e.g., number of elements in the second trace) each backend needs to provide.

The big con with the common format output method is that we're not sure we'll use it elsewhere in the code.

Reply:
I'm not sure I understand what you mean. From what I see the current tests are very generic, that's why I though they could be parametrized across several backends. There would be additionally specific tests for each backend (like for the storage tests) that are not covered by the generic parametrized tests.